go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -241,12 +242,71 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 
 	// Check runtime's whitelist.
 	for _, rt := range paidRuntimes {
-		if rt.AdmissionPolicy.EntityWhitelist != nil && !rt.AdmissionPolicy.EntityWhitelist.Entities[newNode.EntityID] {
-			ctx.Logger().Error("RegisterNode: node's entity not in a runtime's whitelist",
-				"entity", newNode.EntityID,
-				"runtime", rt.ID,
-			)
-			return registry.ErrForbidden
+		if rt.AdmissionPolicy.EntityWhitelist != nil {
```

Comment:
Maybe invert some checks (with continue statements) and get rid of some indentation.